### DISCUSSION TITLE
关于在启动时报错10754端口被占用却找不到占用进程的解决

### DISCUSSION THREAD

**Discussion by MeowSalty** [qier222/YesPlayMusic#1142] [Show and tell]
<img width="361" alt="Snipaste_2021-12-25_16-21-23" src="https://user-images.githubusercontent.com/7829265/147381298-03c7a01e-b72b-40bb-be27-4be55c5b5d17.png">

其实以前用ADB的时候也经常遇到这种情况，不过那会用环境变量修改ADB端口来勉强解决了。那时我猜测估计是QQ之类的软件偷偷占了我的端口，但这次又遇到了这个诡异的问题于是去找了一下，最后发现了这个帖子：[hyper v - What is Administered port exclusions in windows 10? - Stack Overflow](https://stackoverflow.com/questions/58216537/what-is-administered-port-exclusions-in-windows-10)

于是过去一看好家伙果然是Hyper-V吃掉了我的端口（内鬼竟是我自己）:

[这里有一个不存在的图片]

由于不可能总是端口被占了之后去重启一下电脑，因此需要把这个端口加入到Hyper-V的排除端口号内。

```shell
#禁用Hyper-V
dism.exe /Online /Disable-Feature:Microsoft-Hyper-V
#禁用完成后会提示你重启电脑
#排除10754端口
netsh int ipv4 add excludedportrange protocol=tcp startport=10754 numberofports=1
#排除5037端口
netsh int ipv4 add excludedportrange protocol=tcp startport=5037 numberofports=1
#重新启用Hyper-V
dism.exe /Online /Enable-Feature:Microsoft-Hyper-V /All
```

（上面的命令需要以管理员身份执行）

之后这两个端口就不会被Hyper-V占用了。

**Comment by axuanran** [comment]
src/electron/services.js
![Screenshot_2025-10-06-19-33-39-64_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/user-attachments/assets/d2bc196a-9cb7-49ff-93d2-c87c53aa8479)

也可以把这里的端口改掉后修改一下.env 处的地址！
（然后重新编译一下即可，可以拿 github action 编译 ）

https://github.com/qier222/YesPlayMusic/blob/master/.github/workflows/build.yaml
![Screenshot_2025-10-06-19-51-44-06_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/user-attachments/assets/831a9f56-82a5-4a0f-b907-38846a2be073)
如果用 github 编译的话 env 在这里
